unit-tests/func/reflectivity/l500/test-auto-set-options.cpp

```diff
@@ -0,0 +1,192 @@
+//       * If MUR option was already on when IR Reflectivity was turned on,
+//         turning off IR Reflectivity should not turn off MUR option.
+//       * While streaming, Activating IR Reflectivity while Max Usable Range is off
+//         should turn on Max Usable Range option.
+//         If the MUR option was turned on by activating IR Reflectivity,
+//         Turning off IR Reflectivity should turn off MUR option.
+//       * Verify turning off MUR option turns off IR Reflectivity option,
+//         Reflectivity require MUR to be on otherwise it can't work.
+//
+// Note: * L515 specific test
+//       * Test will pass if no L515 device is connected.
+//       * Test will be performed on the first L515 device detected.
+
+TEST_CASE( "IR Reflectivity option change visual preset and sensor mode if not streaming", "[reflectivity]" )
+{
+    auto ds
+        = find_first_supported_depth_sensor_or_exit( "L515", RS2_OPTION_ENABLE_IR_REFLECTIVITY );
+
+    // Verify alt IR is off
+    if( ds.supports( RS2_OPTION_ALTERNATE_IR ) )
+        ds.set_option( RS2_OPTION_ALTERNATE_IR, 0 );
+
+    ds.set_option( RS2_OPTION_SENSOR_MODE, RS2_SENSOR_MODE_XGA );
+    ds.set_option( RS2_OPTION_VISUAL_PRESET, RS2_L500_VISUAL_PRESET_SHORT_RANGE );
+    CHECK_NOTHROW( ds.set_option( RS2_OPTION_ENABLE_IR_REFLECTIVITY, 1.f ) );
+    CHECK( RS2_L500_VISUAL_PRESET_MAX_RANGE == ds.get_option( RS2_OPTION_VISUAL_PRESET ) );
+    CHECK(RS2_SENSOR_MODE_VGA == ds.get_option(RS2_OPTION_SENSOR_MODE));
+}
+
+TEST_CASE("Starting streaming non VGA sensor mode turns off IR Reflectivity", "[reflectivity]")
+{
+    auto ds
+        = find_first_supported_depth_sensor_or_exit("L515", RS2_OPTION_ENABLE_IR_REFLECTIVITY);
+
+    // Verify alt IR is off
+    if (ds.supports(RS2_OPTION_ALTERNATE_IR))
+        ds.set_option(RS2_OPTION_ALTERNATE_IR, 0);
+
+    ds.set_option(RS2_OPTION_VISUAL_PRESET, RS2_L500_VISUAL_PRESET_MAX_RANGE);
+    ds.set_option(RS2_OPTION_SENSOR_MODE, RS2_SENSOR_MODE_VGA);
+
+    // Verify activating IR Reflectivity
+    CHECK_NOTHROW(ds.set_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY, 1.f));
+    CHECK(ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY));
+
+
+    auto profiles = ds.get_stream_profiles();
+    // Find a profile that sensor mode != VGA
+    auto non_vga_depth_profile
+        = std::find_if(profiles.begin(), profiles.end(), [](rs2::stream_profile sp) {
+        auto vp = sp.as< rs2::video_stream_profile >();
+
+        return ((sp.stream_type() == RS2_STREAM_DEPTH) && (sp.format() == RS2_FORMAT_Z16)
+            && (vp.width() != 640) && (vp.height() != 480));
+            });
+
+    REQUIRE(non_vga_depth_profile != profiles.end());
+    ds.open(*non_vga_depth_profile);
+    ds.start([](rs2::frame f) {});
+
+    // Verify turned off
+    CHECK_FALSE(ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY));
+
+    ds.stop();
+    ds.close();
+}
+
+TEST_CASE("Automatically turn on MUR option - stays on", "[reflectivity]")
```

Comment:
I think this test differs from the next because it's NON-streaming?

Reply:
👍